tests/src/JIT/Performance/CodeQuality/HWIntrinsic/X86/PacketTracer/VectorPacket.cs

```diff
@@ -0,0 +1,169 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+
+using static System.Runtime.Intrinsics.X86.Avx;
+using static System.Runtime.Intrinsics.X86.Sse;
+using System.Runtime.Intrinsics;
+using System.Runtime.CompilerServices;
+using System;
+
+internal class VectorPacket256
+{
+    public Vector256<float> Xs;
+    public Vector256<float> Ys;
+    public Vector256<float> Zs;
+    public Vector256<float> Lengths
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        get
+        {
+            var x2 = Multiply(Xs, Xs);
+            var y2 = Multiply(Ys, Ys);
+            var z2 = Multiply(Zs, Zs);
+
+            var l2 = Add(x2, y2);
+            l2 = Add(l2, z2);
+            return Sqrt(l2);
+        }
+    }
+
+
+    public readonly static int Packet256Size = 8;
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public VectorPacket256(Vector256<float> init)
+    {
+        Xs = init;
+        Ys = init;
+        Zs = init;
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public VectorPacket256(float xs, float ys, float zs)
+    {
+        Xs = SetAllVector256(xs);
+        Ys = SetAllVector256(ys);
+        Zs = SetAllVector256(zs);
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public VectorPacket256(Vector256<float> _Xs, Vector256<float> _ys, Vector256<float> _Zs)
+    {
+        Xs = _Xs;
+        Ys = _ys;
+        Zs = _Zs;
+    }
+
+    // Convert AoS vectors to SoA Packet256
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public unsafe VectorPacket256(float* vectors)
+    {
+        Vector256<float> m03 = ExtendToVector256<float>(LoadVector128(&vectors[0])); // load lower halves
+        Vector256<float> m14 = ExtendToVector256<float>(LoadVector128(&vectors[4]));
+        Vector256<float> m25 = ExtendToVector256<float>(LoadVector128(&vectors[8]));
+        m03 = InsertVector128(m03, &vectors[12], 1);  // load higher halves
+        m14 = InsertVector128(m14, &vectors[16], 1);
+        m25 = InsertVector128(m25, &vectors[20], 1);
+
+        var xy = Shuffle(m14, m25, 2 << 6 | 1 << 4 | 3 << 2 | 2);
```

Comment:
We really need a helper intrinsic for constructing the shuffle immediate....
C/C++ has a `__MM_SHUFFLE` macro...

CC. @eerhardt, @CarolEidt